src/os/bluestore/BlueStore.cc

```diff
@@ -9195,12 +9292,28 @@ int BlueStore::_fsck(BlueStore::FSCKDepth depth, bool repair)
       depth == FSCK_SHALLOW ? " (shallow)" : " (regular)")
     << dendl;
 
+  if (repair) {
+    string p = path + "/block";
+    bluestore_bdev_label_t label;
+    _read_bdev_label(cct, p, &label, true);
+    _replicate_bdev_label(p, label, false);
+  }
+
   // in deep mode we need R/W write access to be able to replay deferred ops
   const bool read_only = !(repair || depth == FSCK_DEEP);
   int r = _open_db_and_around(read_only);
   if (r < 0) {
     return r;
   }
+
+  for (uint64_t offset : bdev_label_offsets) {
+    if (offset >= bdev->get_size()) {
+      break;
+    }
+    _unalloc_bdev_label_offset(offset);
+  }
+
+
   auto close_db = make_scope_guard([&] {
```

Comment:
At location: https://github.com/ceph/ceph/pull/53095/files#diff-6f3b1a5dcf5313dc06f71ba4997ee28ec219ecb5633a728b916465c684f1d87dL9308
is the place where the bdev replicas should be marked.